falcon/hooks.py

```diff
@@ -18,8 +18,19 @@
 from inspect import getmembers
 from inspect import iscoroutinefunction
 import re
-
+from typing import Any
+from typing import Awaitable
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Tuple
+from typing import Union
+
+from falcon.asgi.request import Request as AsynchronousRequest
```

Comment:
underscore these `import as` since otherwise they mean that these import are public exports from this module

Reply:
yes, or use `from falcon import asgi` and use `asgi.Request` to type stuff